File: backend/src/main/java/com/jobnest/backend/modules/jobs/domain/Job.java
package com.jobnest.backend.modules.jobs.domain;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

import com.jobnest.backend.modules.company.domain.Company;
import com.jobnest.backend.shared.domain.Account;


@Entity
@Table(name = "jobs")
@Data
public class Job {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "employer_id", nullable = false)
    private Long employerId;

    @Column(name = "company_id")
    private Long companyId;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "category_id", nullable = false)
    private Long categoryId;

    @Column(nullable = false, length = 150)
    private String location;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private JobType type;

    @Column(name = "min_salary")
    private Integer minSalary;

    @Column(name = "max_salary")
    private Integer maxSalary;

    @Column(length = 50)
    private String experience;

    @Column(name = "experience_level", length = 50)
    private String experienceLevel;

    @Column(length = 100)
    private String education;

    @Column(columnDefinition = "TEXT")
    private String skills;

    @Column(name = "is_urgent")
    private Boolean isUrgent = false;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private JobStatus status = JobStatus.ACTIVE;

    @Column(name = "posted_at", nullable = false, updatable = false)
    private LocalDateTime postedAt = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @Column(name = "expires_at")
    private LocalDateTime expiresAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employer_id", insertable = false, updatable = false)
    private Account employer;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", insertable = false, updatable = false)
    private Company company;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id", insertable = false, updatable = false)
    private JobCategory category;

    public enum JobType {
        FULLTIME, PARTTIME, INTERNSHIP, CONTRACT;

    }

    public enum JobStatus {
        ACTIVE, HIDDEN, EXPIRED
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/shared/types/job.ts
export type JobType = "FULLTIME" | "PARTTIME" | "INTERNSHIP" | "CONTRACT";

export type JobStatus =
  | "PENDING_REVIEW"
  | "ACTIVE"
  | "REJECTED"
  | "HIDDEN"
  | "EXPIRED";

export interface Job {
  id: number;
  employerId: number;
  employerName?: string;
  companyId?: number;
  companyName?: string;
  companyLogo?: string;
  title: string;
  description: string;
  categoryId: number;
  categoryName?: string;
  categorySlug?: string;
  location: string;
  type: JobType | string;
  minSalary?: number;
  maxSalary?: number;
  experience?: string;
  experienceLevel?: string;
  education?: string;
  skills?: string;
  isUrgent?: boolean;
  status: JobStatus | string;
  postedAt?: string;
  updatedAt?: string;
  expiresAt?: string;
  viewCount?: number;
  isSaved?: boolean;
}

export interface JobCategory {
  id: number;
  name: string;
  slug?: string;
  iconUrl?: string;
  description?: string;
  openPositions?: number;
}

export interface JobRequest {
  companyId?: number;
  title: string;
  description: string;
  categoryId: number;
  location: string;
  type: JobType;
  minSalary?: number;
  maxSalary?: number;
  experience?: string;
  experienceLevel?: string;
  education?: string;
  skills?: string;
  isUrgent?: boolean;
}